Kernel/Syscalls/mmap.cpp

```diff
@@ -367,7 +367,67 @@ KResultOr<int> Process::sys$mprotect(Userspace<void*> addr, size_t size, int pro
         return 0;
     }
 
-    // FIXME: We should also support mprotect() across multiple regions. (#175) (#964)
+    if (const auto& regions = space().find_regions_intersecting(range_to_mprotect); regions.size()) {
+        size_t full_size_found = 0;
+        // first check before doing anything
+        for (const auto* region : regions) {
+            if (!region->is_mmap())
```

Comment:
I would expect to see a check similar to:

```c++
        // first check before doing anything that all the intersecting regions are mapped from the same vmobject
        const VMObject* vm_obj = nullptr;
        for (const auto* region : regions) {
            if (!vm_obj) {
                vm_obj = &region->vmobject();
            } else if (&region->vmobject() != vm_obj) {
                return EINVAL;
            }
```
to make sure that all the regions we're trying to potentially merge or change perms of were all from the same vmobject originally.

Reply:
I just looked at the [manpage](https://man7.org/linux/man-pages/man2/mprotect.2.html) again, and it doesn't really seem to specifiy, that you aren't allowed to do that, and due to us not merging anything we don't have to really care, can you look up if we have to have the same vmobject across all regions?

Sidenote: We are returning the wrong code on one or more not mapped pages (EINVAL->ENOMEM)

Reply:
All right, I think it's probably up to the OS then. Doing what Linux does until there's a good reason not to seems fine to me.